src/clj/puppetlabs/trapperkeeper/services/protocols/filesystem_watch_service.clj

```diff
@@ -6,9 +6,15 @@
   "Schema for an event on a file watched by this service."
   (schema/if #(= (:type %) :unknown)
     {:type (schema/eq :unknown)
-     :path (schema/pred nil?)}
+     :count schema/Int
+     :watched-path File
+     :changed-path (schema/pred nil?)
+     :full-path (schema/pred nil?)}
```

Comment:
> wrt (schema/maybe File). If I understand that correctly means nil or File, 

right 

> ... which isn't the possible valid values IF we're limiting this to what is valid for :unknown events.

what else can it be?

> I'm in favor of saying that if the event type is :unknown the values must be nil, if the event type is not :unknown they must be a File.

I like this approach but I would rather we just omit the keys entirely instead of enforcing `nil` values for them.

> OR, we don't differentiate between event types and just say that the keys in question may be nil

Yeah, we could go this way but it'd be nice to not pollute the "happy path" schema since the unknown events are so different and funky.